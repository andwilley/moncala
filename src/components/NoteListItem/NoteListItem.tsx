import React from 'react';
import { INote } from '../../api/models';
import Icon from '../../Icon/Icon';

export interface INoteListItemProps {
  note: INote;
  isActive: boolean;
  onDelete: (e: React.MouseEvent) => void;
  onClick: (e: React.MouseEvent) => void;
}

export const NoteListItem: React.FC<INoteListItemProps> = ({
  note,
  onClick,
  onDelete,
  isActive
}) => {
  const content = note.content;
  const lines = content.split(/\r\n|\r|\n/);
  const filledLines = lines.filter(line => /\S/.test(line));

  return (
    <li
      onClick={onClick}
      title="NoteItem"
      style={{ minHeight: '2rem' }}
      className={`relative bb br1 ph2 b--light-gray pv2 hide-child dim pointer ${
        isActive ? 'bg-lightest-blue' : ''
      }`}
    >
      <p className="ma0 truncate f6 b">{filledLines[0]}</p>
      <p className="mt1 mb0 truncate f6 fw5">{filledLines.length > 1 ? filledLines[1] : ''}</p>
      <span className="absolute" style={{ right: 0, top: '.3rem' }} onClick={onDelete}>
        <Icon className="child" size={20} name="delete" />
      </span>
    </li>
  );
};

export default NoteListItem;
