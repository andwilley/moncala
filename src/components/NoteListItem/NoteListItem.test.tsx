import React from 'react';
import renderer from 'react-test-renderer';
import NoteListItem from './NoteListItem';

describe('<NoteListItem> test a basic render output', () => {
  const note = {
    id: 0,
    content: 'test',
    dateCreated: '',
    dateModified: ''
  };
  it('should render the notepad inactive', () => {
    const tree = renderer
      .create(
        <NoteListItem
          note={note}
          isActive={false}
          onClick={() => undefined}
          onDelete={() => undefined}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the notepad active', () => {
    const tree = renderer
      .create(
        <NoteListItem
          note={note}
          isActive={true}
          onClick={() => undefined}
          onDelete={() => undefined}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
