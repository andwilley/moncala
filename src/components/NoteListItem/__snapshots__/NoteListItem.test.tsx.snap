// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteListItem> test a basic render output should render the notepad active 1`] = `
<li
  className="relative bb br1 ph2 b--light-gray pv2 hide-child dim pointer bg-lightest-blue"
  onClick={[Function]}
  style={
    Object {
      "minHeight": "2rem",
    }
  }
>
  <p
    className="ma0 truncate f6 b"
  >
    test
  </p>
  <p
    className="mt1 mb0 truncate f6 fw5"
  >
    
  </p>
  <span
    className="absolute"
    onClick={[Function]}
    style={
      Object {
        "right": 0,
        "top": ".3rem",
      }
    }
  >
    <span
      className="child"
      onClick={[Function]}
    >
      <svg
        className=""
        height={20}
        style={
          Object {
            "fill": "black",
          }
        }
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </span>
  </span>
</li>
`;

exports[`<NoteListItem> test a basic render output should render the notepad inactive 1`] = `
<li
  className="relative bb br1 ph2 b--light-gray pv2 hide-child dim pointer "
  onClick={[Function]}
  style={
    Object {
      "minHeight": "2rem",
    }
  }
>
  <p
    className="ma0 truncate f6 b"
  >
    test
  </p>
  <p
    className="mt1 mb0 truncate f6 fw5"
  >
    
  </p>
  <span
    className="absolute"
    onClick={[Function]}
    style={
      Object {
        "right": 0,
        "top": ".3rem",
      }
    }
  >
    <span
      className="child"
      onClick={[Function]}
    >
      <svg
        className=""
        height={20}
        style={
          Object {
            "fill": "black",
          }
        }
        viewBox="0 0 24 24"
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </span>
  </span>
</li>
`;
